sematic/resolvers/silent_resolver.py

```diff
@@ -16,18 +17,44 @@ class SilentResolver(StateMachineResolver):
     def _schedule_future(self, future: AbstractFuture) -> None:
         self._run_inline(future)
 
+    def _activate_resources(self, future):
+        updated_resources = []
+        for resource in future.props.external_resources:
+            resource = resource.activate(is_local=True)
+            while not (
+                resource.status.state == ResourceState.ACTIVE
+                or resource.status.state.is_terminal()
```

Comment:
Currently a direct transition from ACTIVE to a terminal state is not legal, and indirect transitions from ACTIVE to a terminal state is not possible without calling `_deactivate`.

Why accept `is_terminal` here?

Reply:
adding this comment:
```
# If the resource fails to activate, it may go from
# activating -> deactivating -> deactivated
```